mindsdb/integrations/handlers/notion_handler/notion_handler.py

```diff
@@ -0,0 +1,219 @@
+import os
+import time
+import pandas as pd
+from collections import defaultdict
+
+from pydantic import BaseModel, Extra
+from notion_client import Client
+
+from mindsdb.utilities import log
+from mindsdb.utilities.config import Config
+from mindsdb.integrations.libs.api_handler import APIHandler
+from mindsdb.integrations.libs.response import HandlerStatusResponse as StatusResponse
+
+from .notion_table import (
+    NotionBlocksTable,
+    NotionCommentsTable,
+    NotionDatabaseTable,
+    NotionPagesTable,
+)
+
+
+class NotionHandlerArgs(BaseModel):
+    target: str = None
+    ft_api_info: dict = None
+    ft_result_stats: dict = None
+    runtime: str = None
+    notion_api_token: str = None
+
+    class Config:
+        # for all args that are not expected, raise an error
+        extra = Extra.forbid
+
+
+class NotionHandler(APIHandler):
+    name = "notion"
+
+    def __init__(self, name: str, **kwargs):
+        """constructor
+        Args:
+            name (str): the handler name
+        """
+        super().__init__(name)
+
+        args = kwargs.get("connection_data", {})
+        self.connection_args = {}
+
+        handler_config = Config().get("notion_handler", {})
```

Comment:
Can you please explain what this block of code (lines 47 to 59) is for? I do not quite understand it.

Reply:
It is for setting either the `notion_api_token` or `api_token` as the `connection_args` config. The user can provide any of them, so we need to set them both instead of forcing one of them.

Reply:
Let's not do this. I don't really see a reason to allow two options for the same parameter, let's keep it as `api_token`.

Reply:
Let's also remove the logic for attempting to access the parameters from environment variables or the `Config`, and make it so that the user will need to pass the parameters in when executing a `CREATE DATABASE` statement.